### DISCUSSION TITLE
WiFi Event-based interactions

### DISCUSSION THREAD

**Discussion by HamzaHajeir** [earlephilhower/arduino-pico#2170] [Q&A]
Hi there,

I'm porting out a project called [H4Plugins](https://github.com/hamzahajeir/h4plugins) into supporting RP2040, its WiFi management depends on Event system provided by both Arduino [ESP32](https://github.com/espressif/arduino-esp32/blob/82c36a2fcc43925973fc915b42ad4acd8d1314dd/libraries/WiFi/src/WiFiGeneric.h#L77-L82) and [ESP8266](https://github.com/esp8266/Arduino/blob/685f2c97ff4b3c76b437a43be86d1dfdf6cb33e3/libraries/ESP8266WiFi/src/ESP8266WiFiGeneric.h#L71-L82). 

Couldn't find a reference here for such functionality. 

Is it to be supported soon? and any suggestion/recommendation?

With Thanks,
Hamza hajeir

**Comment by earlephilhower** [comment]
Those are Espressif proprietary interfaces and unfortunately the CYW43 driver doesn't provide anything like that capability now.

**Reply by HamzaHajeir** [reply]
I see, so the way out to keep the system intact is by mimicking the event system by a mechanism that employs polling.

**Reply by earlephilhower** [reply]
Yeah, I think that's the only way to get a lot of those events.  You could *possibly* hack into LWIP to catch link up and down events, but things on the radio (i.e. new STA attached/detached to/from the local AP) are completely invisible to the Pico. :(

**Reply by HamzaHajeir** [reply]
I see, thanks for your enlightment. Fortunately my need is limited to connected/disconnected which is easily achievable. 

Will feedback soon :)
